tests/components/google_photos/test_config_flow.py

```diff
@@ -0,0 +1,191 @@
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json={
+            "refresh_token": "mock-refresh-token",
+            "access_token": "mock-access-token",
+            "type": "Bearer",
+            "expires_in": 60,
+        },
+    )
+
+    setup_api.return_value.mediaItems.return_value.list = Mock()
+    setup_api.return_value.mediaItems.return_value.list.return_value.execute.side_effect = HttpError(
+        Response({"status": "403"}),
+        bytes(load_fixture("google_photos/api_not_enabled_response.json"), "utf-8"),
+    )
+    result = await hass.config_entries.flow.async_configure(result["flow_id"])
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "access_not_configured"
+    assert result["description_placeholders"]["message"].endswith(
+        "Google Photos API has not been used in project 0 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/library/photoslibrary.googleapis.com/overview?project=0 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry."
+    )
+
+
+@pytest.mark.usefixtures("current_request_with_host", "setup_credentials")
+async def test_general_exception(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Check flow aborts if exception happens."""
+    result = await hass.config_entries.flow.async_init(
+        "google_photos", context={"source": config_entries.SOURCE_USER}
+    )
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+    assert result["url"] == (
+        f"{OAUTH2_AUTHORIZE}?response_type=code&client_id={CLIENT_ID}"
+        "&redirect_uri=https://example.com/auth/external/callback"
+        f"&state={state}"
+        "&scope=https://www.googleapis.com/auth/photoslibrary.readonly"
+        "+https://www.googleapis.com/auth/userinfo.profile"
+        "&access_type=offline&prompt=consent"
+    )
+
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json={
+            "refresh_token": "mock-refresh-token",
+            "access_token": "mock-access-token",
+            "type": "Bearer",
+            "expires_in": 60,
+        },
+    )
+
+    with patch(
+        "homeassistant.components.google_photos.api.build",
+        side_effect=Exception,
+    ):
+        result = await hass.config_entries.flow.async_configure(result["flow_id"])
+
+    assert result["type"] is FlowResultType.ABORT
```

Comment:
Why abort though?

Reply:
I assume you're implying that instead of aborting you might expect this to render an error message and let the user re-run the step?

Haven't thought about this much since this is how every current implementation of `async_oauth_create_entry` behaves.

I suspect they all work this way because this page is shown after a redirect to an external website. Starting the config flow over might just result in another redirect? Note there are no steps really in the google photos config flow.

My assumption is that if they hit an exception it probably means something in setup is wrong and it won't succeed if they try again.

Let me know if i've misunderstood and there is something else you're getting at here.

Reply:
Oh no you're correct, there's no real way to redo this step